common/src/main/java/uk/gov/gchq/palisade/service/request/DataRequestResponse.java

```diff
@@ -83,6 +90,15 @@ public void setResources(final Map<LeafResource, ConnectionDetail> resources) {
         return resources;
     }
 
+    public void setOriginalRequestId(final String originalRequestId) {
+        this.originalRequestId = originalRequestId;
+    }
+
```

Comment:
Not sure I understand the point you're making??
originalRequestId needs to be included as part of the base level request.
If a caller makes use of the originalRequestId then we need to throw an exception.
Perhaps we could discuss this point further please???

Reply:
the originalRequestId() method contains a null check and could contain other validity checks. In the rest of the repository, we have used the convention where the "raw" version of the setter does all the checks and mutations and then returns this as in this class. The "set" variant of a method just calls the method and ignores the return value. Thus the same validation is done in both cases.

In this case, the "set" variant is missing any validation and would allow originalRequestID to be set to null. Rather than duplicate the validation logic, just call originalRequestId() from inside the "set' method. For examples, look at any Request subclass.

Reply:
Ah never mind the change has been made since I read that last comment.